dev/run-python-sagemaker-tests.sh

```diff
@@ -22,8 +22,8 @@ pytest tests/sagemaker/mock --large
 # TODO: split out sagemaker tests and move other spark tests to run-python-flavor-tests.sh
 pytest tests/spark --large  --ignore tests/spark/test_mleap_model_export.py
 
-# MLeap doesn't support spark 3.x (https://github.com/combust/mleap#mleapspark-version)
-pip install pyspark==2.4.5
+# MLeap doesn't support spark > 3.0.2: https://github.com/combust/mleap#mleapspark-version
+pip install pyspark==3.0.2 mleap
```

Comment:
A historic day! MLeap now supports PySpark? Does that mean we can revive https://github.com/mlflow/mlflow/pull/4721?

Reply:
Looks like it does:

https://github.com/combust/mleap#mleapspark-version

![image](https://user-images.githubusercontent.com/17039389/133194906-6a2df113-2396-44a4-b377-ee89da05a801.png)

Yep, I think we can!

Reply:
Cool! Can we do that as part of this PR?

Reply:
@dbczumar Updated the RP!